src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -50,120 +51,108 @@ public Result charTyped(char charTyped,
+                            onTypeFormattingSettings.formatOnCloseBraceScope = formatOnCloseBraceScope;
+                        }
+                        onTypeFormattingSettings.formatOnCloseBraceCharacters += formattingFeature.getFormatOnCloseBraceCharacters(file);
 
-            // Close braces
-            if (formattingFeature.isFormatOnCloseBrace(file) &&
-                // Make sure the formatter supports formatting of the configured scope
-                ((formattingFeature.getFormatOnCloseBraceScope(file) == FormattingScope.FILE) || rangeFormattingSupported)) {
-                Map.Entry<Character, Character> bracePair = ContainerUtil.find(
-                        LSPIJEditorUtils.getBracePairs(file).entrySet(),
-                        entry -> entry.getValue() == charTyped
-                );
-                if (bracePair != null) {
-                    String formatOnCloseBraceCharacters = formattingFeature.getFormatOnCloseBraceCharacters(file);
-                    Character openBraceChar = bracePair.getKey();
-                    Character closeBraceChar = bracePair.getValue();
-                    if (StringUtil.isEmpty(formatOnCloseBraceCharacters) ||
-                        (formatOnCloseBraceCharacters.indexOf(closeBraceChar) > -1)) {
-                        return handleCloseBraceTyped(
-                                project,
-                                editor,
-                                file,
-                                formattingFeature,
-                                openBraceChar,
-                                closeBraceChar
-                        );
+                        onTypeFormattingSettings.formatOnStatementTerminator |= formattingFeature.isFormatOnStatementTerminator(file);
+                        FormattingScope formatOnStatementTerminatorScope = formattingFeature.getFormatOnStatementTerminatorScope(file);
+                        if (formatOnStatementTerminatorScope.compareTo(onTypeFormattingSettings.formatOnStatementTerminatorScope) > 0) {
+                            onTypeFormattingSettings.formatOnStatementTerminatorScope = formatOnStatementTerminatorScope;
+                        }
+                        onTypeFormattingSettings.formatOnStatementTerminatorCharacters += formattingFeature.getFormatOnStatementTerminatorCharacters(file);
+
+                        onTypeFormattingSettings.formatOnCompletionTrigger |= formattingFeature.isFormatOnCompletionTrigger(file);
+                        onTypeFormattingSettings.formatOnCompletionTriggerCharacters += formattingFeature.getFormatOnCompletionTriggerCharacters(file);
                     }
                 }
-            }
+        );
+        boolean rangeFormattingSupported = rangeFormattingSupportedRef.get();
 
-            // Statement terminators
-            if (formattingFeature.isFormatOnStatementTerminator(file) &&
+        // Close braces
+        if (onTypeFormattingSettings.formatOnCloseBrace &&
```

Comment:
And everything below here is just updating to use that collected info. I recommend hiding whitespace-only changes for the best diff.